test/index.test.ts

```diff
@@ -68,8 +68,11 @@ test('Should ensure formatting is correct for PublicInputs', async function (ass
 test('Should ensure formatting is correct for extracted PublicInputs', async function (assert) {
   // For each circuit
   testVectors.forEach((testVector, i) => {
+    // Ensure testVector proof exists
+    assert.ok(testVector.proof, `Test vector ${i} is missing a proof, which is required for this test`)
+
     // Extract PublicInputs from existing CircuitInputs
-    const returnedPublicInputs = extractPublicInputsFromCircuitInputs(testVector.inputs, testVector.proof)
+    const returnedPublicInputs = extractPublicInputsFromCircuitInputs(testVector.inputs, testVector.proof as Proof)
```

Comment:
is the `as Proof` required here? Isn't it already that type?

Reply:
`Proof` can be undefined in the testVectors, as it is not needed for the first `prove()` call, but is returned by a `prove()` call. Basically, I didn't want to require it in the testVectors since it isn't needed until calling `verify()`. 

```
Argument of type 'Proof | undefined' is not assignable to parameter of type 'Proof'.
  Type 'undefined' is not assignable to type 'Proof'.
```
  
The typescript compiler doesn't accept the `assert.ok` statement above line 75, that checks for `testVectors.proof` existence, so I force the type immediately below the check.